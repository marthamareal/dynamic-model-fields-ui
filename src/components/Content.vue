<template>
<div class="row" id="content">
  <div v-if="riskTypes.length === 0" class="d-flex justify-content-center mb-3">
    <b-spinner variant="success"></b-spinner>
    <strong>Loading...</strong>
  </div>

  <div class="row cards-container" id="risk-types" v-else >
    <Card :riskType="riskType" v-for="riskType in riskTypes" :key="riskType.indexOf"/>
  </div>
  <div class="row align-self-center " v-if="riskTypes.length > 0" >
    <b-pagination
      v-model="currentPage"
      :per-page="perPage"
      :items="riskTypes"
      :total-rows="totalItems"
      @change="pageChange"
      >
    </b-pagination>
  </div>
</div>
</template>

<script>
import axios from 'axios'
import Card from './Card'

export default {
  name: 'Content',
  components: {
    Card
  },
  data () {
    return {
      perPage: 6,
      currentPage: 1,
      riskTypes: [],
      totalItems: null
    }
  },
  methods: {
    handleApiErrors () {
      this.$bvToast.toast(` Unable to fetch risk types, Please check your internet connections or contact our support for help.`, {
        title: 'Sorry',
        toaster: 'b-toaster-top-center',
        variant: 'danger',
        solid: true
      })
    },
    async pageChange (page) {
      this.currentPage = page
      await axios.get(`${process.env.BASE_API_URL}/risk-types/?page=${page}`)
        .then(response => { this.riskTypes = response.data.results; this.totalItems = response.data.count })
        .catch(response => {
          this.handleApiErrors()
        })
    }
  },
  async mounted () {
    await axios.get(`${process.env.BASE_API_URL}/risk-types/`)
      .then(response => { this.riskTypes = response.data.results; this.totalItems = response.data.count })
      .catch(response => {
        this.handleApiErrors()
      })
  }
}
</script>

<style>
  #content {
      margin: 0;
      padding: 30px 60px 0px 60px;
      min-height: 70vh;
      flex-direction: column;
  }
  .cards-container {
    flex-grow: 1
  }
</style>
