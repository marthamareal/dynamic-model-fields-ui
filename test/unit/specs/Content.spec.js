import { mount } from '@vue/test-utils'
import Vue from 'vue';

import Content from '@/components/Content.vue'

import BSpinner from 'bootstrap-vue/es/components/spinner/spinner'
import BPagination from 'bootstrap-vue/es/components/pagination/pagination'
import BButton from 'bootstrap-vue/es/components/button/button'
import BModal from 'bootstrap-vue/es/components/modal/modal'
import vBModal from 'bootstrap-vue/es/directives/modal/modal'

Vue.directive('b-modal', vBModal)
Vue.component('b-modal', BModal)
Vue.component('b-button', BButton)
Vue.component('b-pagination', BPagination)
Vue.component('b-spinner', BSpinner)


jest.mock('axios', () => ({
    get: () => {
        const response = { 
            data: {
                results: [ {
                    "id": "10d98576-091f-4fb2-ad14-c40822748b51",
                    "name": "Professional liability insurance",
                    "description": "Professional liability insurance, also known",
                    fileds: [
                        {
                        "id": "4b9fbf08-dd06-4320-b409-dc469c6fcb32",
                        "name": "date",
                        "field_type": "date",
                        "options": null
                        },
                        {
                        "id": "483a996e-e603-4437-b689-99193383a7e8",
                        "name": "name",
                        "field_type": "text",
                        "options": null
                        }
        
                    ]
                }],
                count: 7
            }
        }
        return new Promise((resolve, reject) => {resolve(response)})
    }
  }))

describe('Content.vue', () => {
    const wrapper = mount(Content)
    it('should render as expected', () => {
        wrapper.setData({
            riskTypes: [ {
                "id": "10d98576-091f-4fb2-ad14-c40822748b51",
                "name": "Professional liability insurance",
                "description": "Professional liability insurance, also known",
                fileds: [
                    {
                    "id": "4b9fbf08-dd06-4320-b409-dc469c6fcb32",
                    "name": "date",
                    "field_type": "date",
                    "options": null
                    },
                    {
                    "id": "483a996e-e603-4437-b689-99193383a7e8",
                    "name": "name",
                    "field_type": "text",
                    "options": null
                    }
    
                ]
            }]
        })
        expect(wrapper.html()).toMatchSnapshot();
    })
    it('pagination changes should work as expected', () =>{
        const linkWrapper = wrapper.findAll('.page-link').at(3)
        const refWrapper = wrapper.findAll('.page-item').at(3)
        expect(refWrapper.classes('active')).toBe(false)
        linkWrapper.trigger('click')
        expect(refWrapper.classes('active')).toBe(true)
        expect(wrapper.findAll('.page-item').at(2).classes('active')).toBe(false)
    })
})