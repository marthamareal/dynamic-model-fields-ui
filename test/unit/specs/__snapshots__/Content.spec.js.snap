// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content.vue should render as expected 1`] = `
<div id="content" class="row">
    <div class="row cards-container" id="risk-types">
        <div id="card" class="col-md-4 mb-5">
            <div class="card h-100">
                <div class="card-body">
                    <h2 class="card-title">Professional liability insurance</h2>
                    <p class="card-text">Professional liability insurance, also known</p>
                </div>
                <div class="card-footer"><button type="button" class="btn btn-primary">APPLY</button>
                    <div>
                        <div id="modal-10d98576-091f-4fb2-ad14-c40822748b51___BV_modal_outer_" style="position: absolute; z-index: 1040;">
                            <div id="modal-10d98576-091f-4fb2-ad14-c40822748b51" role="dialog" tabindex="-1" aria-hidden="true" class="modal fade" style="display: none;">
                                <div class="modal-dialog modal-md">
                                    <div role="document" id="modal-10d98576-091f-4fb2-ad14-c40822748b51___BV_modal_content_" aria-labelledby="modal-10d98576-091f-4fb2-ad14-c40822748b51___BV_modal_header_" aria-describedby="modal-10d98576-091f-4fb2-ad14-c40822748b51___BV_modal_body_" class="modal-content">
                                        <header id="modal-10d98576-091f-4fb2-ad14-c40822748b51___BV_modal_header_" class="modal-header">
                                            <h5 class="modal-title">Professional liability insurance</h5><button type="button" aria-label="Close" class="close">×</button>
                                        </header>
                                        <div id="modal-10d98576-091f-4fb2-ad14-c40822748b51___BV_modal_body_" class="modal-body"> <button type="button" class="btn mt-3 btn-primary btn-secondary btn-block">SUBMIT</button></div>
                                        <!---->
                                    </div>
                                </div>
                            </div>
                            <!---->
                            <!---->
                        </div>
                    </div>
                </div>
            </div>
        </div>
    </div>
    <div class="row align-self-center ">
        <ul role="menubar" aria-disabled="false" aria-label="Pagination" class="pagination b-pagination pagination-md" items="[object Object]">
            <li role="none presentation" aria-hidden="true" class="page-item disabled"><span role="menuitem" aria-label="Go to first page" aria-disabled="true" class="page-link">«</span></li>
            <li role="none presentation" aria-hidden="true" class="page-item disabled"><span role="menuitem" aria-label="Go to previous page" aria-disabled="true" class="page-link">‹</span></li>
            <!---->
            <li role="none presentation" class="page-item active"><a role="menuitemradio" aria-label="Go to page 1" aria-checked="true" aria-posinset="1" aria-setsize="2" tabindex="0" target="_self" href="#" class="page-link">1</a></li>
            <li role="none presentation" class="page-item"><a role="menuitemradio" aria-label="Go to page 2" aria-checked="false" aria-posinset="2" aria-setsize="2" tabindex="-1" target="_self" href="#" class="page-link">2</a></li>
            <!---->
            <li role="none presentation" class="page-item"><a role="menuitem" tabindex="-1" aria-label="Go to next page" target="_self" href="#" class="page-link">›</a></li>
            <li role="none presentation" class="page-item"><a role="menuitem" tabindex="-1" aria-label="Go to last page" target="_self" href="#" class="page-link">»</a></li>
        </ul>
    </div>
</div>
`;
